src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,263 @@
+{
+    if (ex.cmd == "new")
+        return CommandTypes::NEW;
+
+    if (ex.cmd == "open" || ex.cmd == "opn")
+        return CommandTypes::OPEN;
+
+    if (ex.cmd == "next" || ex.cmd == "nxt")
+        return CommandTypes::NEXT_TAB;
+
+    if (ex.cmd == "previous" || ex.cmd == "prv")
+        return CommandTypes::LAST_TAB;
+
+    if (ex.cmd == "compile" || ex.cmd == "cmp")
+        return CommandTypes::COMPILE;
+
+    if (ex.cmd == "crun" || ex.cmd == "crn")
+        return CommandTypes::COMPILE_RUN;
+
+    if (ex.cmd == "run")
+        return CommandTypes::RUN;
+
+    if (ex.cmd == "drun" || ex.cmd == "drn")
+        return CommandTypes::DETACHED_RUN;
+
+    if (ex.cmd == "killall" || ex.cmd == "kap")
+        return CommandTypes::KILL_PROCESS;
+
+    if (ex.cmd == "format" || ex.cmd == "fmt")
+        return CommandTypes::FORMAT_CODE;
+
+    if (ex.cmd == "snippet" || ex.cmd == "snp")
+        return CommandTypes::SNIPPETS;
+
+    if (ex.cmd == "vmode" || ex.cmd == "vmd")
+        return CommandTypes::VMODE;
+
+    if (ex.cmd == "preference" || ex.cmd == "prf")
+        return CommandTypes::PREFERENCE;
+
+    if (ex.cmd == "lang" || ex.cmd == "lng")
+        return CommandTypes::CHLANG;
+
+    if (ex.cmd == "clear" || ex.cmd == "clr")
+        return CommandTypes::CLEAR;
+
+    if (ex.cmd == "exit" || ex.cmd == "ext")
+        return CommandTypes::EXIT;
+
+    return CommandTypes::UNKNOWN;
+}
+
+bool FakeVimCommand::handleCustomCommand(CommandTypes type, QString const &args, bool hasbang)
+{
+    if (type == CommandTypes::UNKNOWN)
+        return false;
+
+    switch (type)
+    {
+    case CommandTypes::NEW: {
+        QString lang;
+
+        if (args == "c++" || args == "cpp")
+            lang = "C++";
+        else if (args == "py" || args == "python")
+            lang = "Python";
+        else if (args == "java")
+            lang = "Java";
+        else if (args.isEmpty())
+            lang = SettingsHelper::getDefaultLanguage();
+
+        if (!lang.isEmpty())
+            appwin->openTab("", lang);
+        else
+            showError(tr("new requires empty or one of 'cpp', 'java' and 'python' argument, got %1").arg(args));
+        break;
+    }
+
+    case CommandTypes::NEXT_TAB: {
+        auto total = appwin->totalTabs();
+        auto curr = appwin->indexOfWindow(appwin->currentWindow());
+        int next = (curr + 1) % total;
+        if (next != curr)
+            appwin->setTabAt(next);
+        break;
+    }
+    case CommandTypes::LAST_TAB: {
+        auto total = appwin->totalTabs();
+        auto curr = appwin->indexOfWindow(appwin->currentWindow());
+        int last = curr ? curr - 1 : total - 1;
+        if (last != curr)
+            appwin->setTabAt(last);
+        break;
+    }
+
+    case CommandTypes::OPEN: {
+        QString path = args;
+        if (path.isEmpty())
+        {
+            showError(tr("open requires a file path"));
```

Comment:
I suggest opening the file dialog here.

Reply:
This will be same as clicking open in action menu